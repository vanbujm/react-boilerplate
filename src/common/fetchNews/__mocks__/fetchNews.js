/* eslint-disable import/prefer-default-export */
const mockNewsData = [
  {
    content: '<h1>Some mock content with html </h1>',
    link: 'https://reactjsnews.com/fake-news',
    title: 'Using Proxies with Redux Types',
  },
];
export async function fetchNews() {
  return {
    json: async () => ({ data: { news: mockNewsData } }),
  };
}
