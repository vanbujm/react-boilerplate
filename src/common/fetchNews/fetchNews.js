/* eslint-disable import/prefer-default-export */

export async function fetchNews(fetchFunction) {
  return fetchFunction('/graphql', {
    body: JSON.stringify({
      query: '{news{title,link,content}}',
    }),
  });
}
