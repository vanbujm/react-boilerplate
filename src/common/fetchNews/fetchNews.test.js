/* eslint-env jest */
import '@babel/polyfill';
import { fetchNews } from './fetchNews';

const fetchSpy = jest.fn();

describe('fetchNews', () => {
  test('calls the fetch function', async () => {
    await fetchNews(fetchSpy);
    expect(fetchSpy.mock.calls).toMatchSnapshot();
  });
});
