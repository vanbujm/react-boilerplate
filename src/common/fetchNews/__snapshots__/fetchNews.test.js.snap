// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchNews calls the fetch function 1`] = `
Array [
  Array [
    "/graphql",
    Object {
      "body": "{\\"query\\":\\"{news{title,link,content}}\\"}",
    },
  ],
]
`;
